pkgs/development/mobile/androidenv/examples/shell-with-emulator.nix

```diff
@@ -0,0 +1,150 @@
+{
+  # To test your changes in androidEnv run `nix-shell android-sdk-with-emulator-shell.nix`
+
+  # If you copy this example out of nixpkgs, use these lines instead of the next.
+  # This example pins nixpkgs: https://nix.dev/tutorials/towards-reproducibility-pinning-nixpkgs.html
+  /*nixpkgsSource ? (builtins.fetchTarball {
+    name = "nixpkgs-20.09";
+    url = "https://github.com/NixOS/nixpkgs/archive/20.09.tar.gz";
+    sha256 = "1wg61h4gndm3vcprdcg7rc4s1v3jkm5xd7lw8r2f67w502y94gcy";
+  }),
+  pkgs ? import nixpkgsSource {
+    config.allowUnfree = true;
+  },
+  */
+
+  # If you want to use the in-tree version of nixpkgs:
+  pkgs ? import ../../../../.. {
+    config.allowUnfree = true;
```

Comment:
Can we put the example in a markdowm file to avoid evaluating unfree choice by accident on hydra?

Reply:
i think its fine currently

Reply:
when for the first time I saw the `shell.nix` example, personally, it was helpful for me to get the point. However, if you think it needs more clarification, just let me know more about it. Thanks!

Reply:
The problem is that it cannot be sourced by any other nix file because it would enable unfree packages which we don't want to on hydra.

Reply:
yes! there's no need to source them by any other nix file, except my idea that we could run them in the tests-suit in CI, which is failed because of this exact reason. But in the end, we need to have some examples, and as I mentioned [before](https://github.com/NixOS/nixpkgs/pull/213871#discussion_r1110277358), it can be run only locally. Additionally, there's another `shell.nix` file in the `examples` directory with the same conditions.